app/api/models/region_validation.py

```diff
@@ -67,21 +67,33 @@ def _validate_region_name(self):
                     )
                 else:
                     self._is_valid[0] = True
-
-                    if (self.start > 0) and (self.start < genome_region.length):
-                        self._is_valid[1] = True
-                    else:
+                    try:
```

Comment:
Can't we replace this part with a Pydantic exception handler? I'm not sure if all of this is necessary and having exceptions inside exceptions doesn't look like a good idea unless we have to!

Reply:
> Can't we replace this part with a Pydantic exception handler?
Technically we can but that means entire validation logic has to move there which I think would be too much. Certainly we can move if start/end is int in the validation part but _is_valid[] holds True when 

1. star is int
2. start is within 1 and length ( which you wouldn't know until you have region info)
3. start < end
like wise for end and region.
We can still do this but then we have to have another set of flags which would indicate the validity of the input (if its int or not ) and if it is then only check for the validation of the region.
Moreover, This same region logic is used by another endpoint which just validates the region ( /validate_location/genome/<abc>/location=<Y>). Which makes it even more complicated  

> having exceptions inside exceptions doesn't look like a good idea
This is because requirement is if start is invalid it has to say start is invalid ( same for end ). I can catch it in the outer try but then one has to put in a logic to check what is wrong is it region/start/end so keeping it simple and catch exception where it makes most sense to.